Changes regarding 'shortmess', ":h :echo-redraw", and use of sigils

lifecrisis opened the pull request:
Hi @rburny,

I made some minor improvements to your script. I think it's a great feature, and I've had fun using it at work. I actually think this feature should be implemented in the Vim core, perhaps as an option. That's way down the road, though. You'd have to wade through an ocean of C to get it done.

The reasons for my changes are all contained in the commit messages, themselves.

Thanks for your review, and for the plugin!

Commit:
Modify the 'shortmess' option

When a search motion requires that we wrap around to the beginning or
the end of the buffer, Vim normally prints a message notifying the user
that the search is continuing from the other end of the file. This can
cause a "flicker" when working in gvim. Suppressing this message solves
the problem.

Commit:
Place a :redraw call before the index is printed

By updating the screen prior to echoing the index, we avoid the
possibility of clearing the message with a delayed redrawing operation.
This seems to improve the smoothness of scanning rapidly through
searches in the GUI (e.g., by holding down 'n'). In any case, the change
doesn't hurt any functionality and can be added without worry.

See ":h :echo-redraw".

Commit:
Refactor one function to use sigils properly

Here, I refactored s:PrintMatches() to correctly use the sigil for
function-local identifiers. In general, it is good style to use the
sigils provided by Vim script to make scoping for identifiers clear and
explicit. A future refactor of the entire script to employ this style
convention is recommended.

lifecrisis commented:
**Note:** Please approve my changes if you desire, and I will then sign the Google CLA.

rburny commented on plugin/searchindex.vim:
```diff
@@ -38,6 +38,8 @@ if !exists('g:searchindex_star_case')
   let g:searchindex_star_case=1
 endif
 
+set shortmess+=s
```
Please add a short comment here, it's easier than looking it up in Vim help every time.

rburny commented:
Thanks; these look good. Please sign the CLA, and I'll merge your changes.

Commit:
Add a comment before modifying 'shortmess'

The comment makes the purpose of the setting clear without requiring the
reader to refer to the Vim documentation.

lifecrisis commented:
Hey @rburny, I've signed the CLA using my Github username. Should be all good to go!

rburny commented:
Thank you!
